### ISSUE TITLE
Lambda support

### ISSUE THREAD

**Issue by frohoff** [awslabs/amazon-kinesis-scaling-utils#40]
It would be great if this could be run as a scheduled lambda function instead of on EC2/EB. Though I did notice that there's already a `LambdaApp` branch, I didn't see any obvious or documented way to do this.

**Comment by pgieser** [comment]
I had wondered this as well, but assumed that it was not implemented because any autoscaling wouldn't necessarily finish in the 5 minutes that a lambda has available.  Also, you would have to keep track of a fair amount of state between executions (not impossible, but starts to make it less appealing I suppose).

**Comment by IanMeyers** [comment]
Yeah - today this isn't possible as it would need to run forever. I would happily take a PR for a version that ran a single cycle of the StreamMonitor using CloudWatch Events on a tight schedule, if someone is interested in a hobby project.

**Comment by racheliurui** [comment]
what about prebuilt docker image can run using fargate

**Comment by igilham** [comment]
Old news now, but for the sake of helping newcomers, [a Lambda solution](https://github.com/aws-samples/kinesis-auto-scaling) has been published.
